cinema/views.py

```diff
@@ -94,13 +97,32 @@ def get_queryset(self):
 
         return queryset.distinct()
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        permission_classes=[IsAdminUser],
+        url_path="upload-image"
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
```

Comment:
Ensure that the `get_serializer` method correctly returns `MovieImageSerializer` for the `upload_image` action. This is already handled in `get_serializer_class`, but double-check to ensure consistency.